### ISSUE TITLE
Rule proposal: Prevent setting nested writable computed value

### ISSUE THREAD

**Issue by Armaldio** [vuejs/eslint-plugin-vue#1891]
**Please describe what the rule should do:**
The rule should warn when trying to set a writable computed ref nested property.

**What category should the rule belong to?**

- [ ] Enforces code style (layout)
- [x] Warns about a potential error (problem)
- [ ] Suggests an alternate way of doing something (suggestion)
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```js
import Vue from 'vue'
import { defineComponent, ref, computed } from '@vue/composition-api'

export default defineComponent({
  setup(props, { root }) {
    const value = ref('French')

    const language = computed({
      get() {
        return {
         	selected: value.value 
        }
      },
      set({ selected }) {
        value.value = selected
      }
    })
    
    console.log(value) // French
    console.log(language) // { selected: 'French' }
    
    // Good
    language.value = { selected: 'English' }
    
    // Bad, it has no effect unlike a standard ref
    language.value.selected = 'German'

    return {
	  language
    }
  }
)}
```

**Additional context**

With writable computed ref, it's easy to fall into this case since IDE suggestion will show nested properties.

**Comment by Armaldio** [comment]
Reactivity is working fine in case of a regular `ref`

**Comment by FloEdelmann** [comment]
Thanks for checking. Then I'd suggest rule name `vue/no-deep-mutating-computed-ref`.
